### PR TITLE
Remove Py2 support

### LINKED ISSUE #3612: Python 2.7 deprecation
In 6 months from now `boto3` will finally drop support for Python 2.7:
> Boto3 will no longer support Python 2.7 starting July 15, 2021. To continue receiving service updates, bug fixes, and security updates please upgrade to Python 3.6 or later. More information can be found here: https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-python-2-7-in-aws-sdk-for-python-and-aws-cli-v1/

Python 2.7 reached EOL already several months ago, what are `moto`'s plans on the topic?

### PR DESCRIPTION
**Pull request by bblommers**
Fixes #3612 

I realize it's not July yet, but if we do want to move in this direction, at least we've got a head start.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [ 2.7, 3.6, 3.7, 3.8 ]
+        python-version: [ 3.6, 3.7, 3.8 ]
```
Comment (mikegrima) [inline .github/workflows/build.yml:12]: Let's add 3.9

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -79,7 +78,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [2.7, 3.6, 3.7, 3.8]
+        python-version: [3.6, 3.7, 3.8]
```
Comment (mikegrima) [inline .github/workflows/build.yml:81]: Ditto

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -122,7 +120,7 @@ jobs:
     needs: lint
     strategy:
       matrix:
-        python-version: [2.7, 3.6, 3.7, 3.8]
+        python-version: [3.6, 3.7, 3.8]
```
Comment (mikegrima) [inline .github/workflows/build.yml:123]: Ditto

**Review by mikegrima — APPROVED** [review]
Rebase off the latest master, and if tests pass feel free to merge.

File: .github/workflows/build.yml (hunk 1 above)
Comment (mikegrima) [reply .github/workflows/build.yml:12]: Also remove 3.6

**Comment by bblommers** [comment]
Thanks @mikegrima. I'll leave it open for now until @spulec has a look, IIRC he wanted to support it at least as long as boto3 supports it

File: .github/workflows/build.yml (hunk 1 above)
Comment (bblommers) [reply .github/workflows/build.yml:12]: 3.6 is EOL in 12/2021, so not quite yet.
3.9 support is in this PR: #3848

**Pull request merged**